CMakeLists.txt

```diff
@@ -409,6 +355,10 @@ set(Boost_ADDITIONAL_VERSIONS
 # CMake'.  Good luck!  You'll need it.
 set(Boost_NO_BOOST_CMAKE 1)
 find_package(Boost "${BOOST_VERSION}")
+if(NOT Boost_FOUND AND (NOT DISABLE_DOWNLOADS))
+  get_deps(https://dl.bintray.com/boostorg/release/1.75.0/source/boost_1_75_0.tar.gz boost boost_1_75_0.tar.gz)
+  find_package(Boost REQUIRED)
+endif()
```

Comment:
Hmm, I realized that if the user set `DISABLE_DOWNLOADS`, we might end up not issuing an error if Boost isn't found.  Here's another strategy that will use the default `find_package()` errors:

```
set(Boost_NO_BOOST_CMAKE 1)
if (DISABLE_DOWNLOADS)
  find_package(Boost "${BOOST_VERSION}" REQUIRED)
elseif (NOT DISABLE_DOWNLOADS)
  find_package(Boost "${BOOST_VERSION}")
  if (NOT Boost_FOUND)
    get_deps(https://dl.bintray.com/boostorg/release/1.75.0/source/boost_1_75_0.tar.gz boost boost_1_75_0.tar.gz)
    find_package(Boost REQUIRED)
  endif ()
endif ()
```

Probably we should use that pattern (or similar) for all instances where we use `get_deps()` to get the dependency if it's not available.

Reply:
Totally agreed, otherwise we will run into a situation, where libraries are not found, and the user has disabled download